content/docs/reference/how.md

```diff
@@ -7,44 +7,89 @@ menu:
 
 ```javascript
-const bucket = new aws.s3.Bucket("media-bucket");
-const bucket = new aws.s3.Bucket("app-bucket"); // renamed bucket
+const mediaBucket = new aws.s3.Bucket("media-bucket", {
+    acl: "public-read",   // add acl
+});
+const contentBucket = new aws.s3.Bucket("content-bucket");
 ```
 
-This time, the engine will not create another `media-bucket`, since it exists in the checkpoint. Now, since an S3 bucket cannot be renamed in place, the engine makes a "replace" call to the AWS provider. The provider deletes the bucket `content-bucket125ce` and creates a new one.
+When you run `pulumi preview` or `pulumi up`, the entire process starts over.  The language host starts running your program and the call to aws.s3.Bucket causes a new resource registration request to be sent to the engine. This time, however, our state already contains a resource named `media-bucket`, so engine asks the resource provider to compare the existing state from our previous run of `pulumi up` with the desired state expressed by the program. The process detects that the `acl` property has changed from `private` (the default value) to `public-read`. By again consulting the resource provider the engine determines that it is able to update this property without creating a new bucket, and so it tells the provider to update the acl property to `public-read`. When this operation completes, the current state is updated to reflect the change that had been made.
 
-## Creation and Deletion Order
+The engine also receives a resource registration request for "content-bucket".  However, since there are no changes between the current state and the desired state, the engine does not need to make any changes to the resource.
+
+Now, suppose we rename `content-bucket` to `app-bucket`.
 
-A needed resource will be created or updated once all of the Inputs it depends on are satisfied.
+```javascript
+const mediaBucket = new aws.s3.Bucket("media-bucket", {
+    acl: "public-read",
+});
+const contentBucket = new aws.s3.Bucket("app-bucket"); // renamed bucket
+```
+
+This time, the engine will not need to make any changes to `media-bucket` since its desired state matches its actual state. However, when the resource request for `app-bucket` is processed, the engine sees there's no existing resource named `app-bucket` in the current state and so it must create a new S3 bucket.  Once that process is complete and the language host has shut down, the engine looks for any resources in the current state which it did not see resource registration for. In this case, since we removed the registration of `content-bucket` from our program, the engine calls the resource provider to delete the existing `content-bucket` bucket.
```

Comment:
```suggestion
This time, the engine will not need to make any changes to `media-bucket` since its desired state matches its actual state. However, when the resource request for `app-bucket` is processed, the engine sees there's no existing resource named `app-bucket` in the current state and so it must create a new S3 bucket.  Once that process is complete and the language host has shut down, the engine looks for any resources in the current state which it did not see a resource registration for. In this case, since we removed the registration of `content-bucket` from our program, the engine calls the resource provider to delete the existing `content-bucket` bucket.
```